src/main/java/io/codelex/flightplanner/services/FlightDbService.java

```diff
@@ -0,0 +1,91 @@
+package io.codelex.flightplanner.services;
+
+import io.codelex.flightplanner.api.AddFlightRequest;
+import io.codelex.flightplanner.api.PageResult;
+import io.codelex.flightplanner.api.SearchFlightRequest;
+import io.codelex.flightplanner.domain.Airport;
+import io.codelex.flightplanner.domain.Flight;
+import io.codelex.flightplanner.repositories.FlightDbRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class FlightDbService extends FlightService {
+
+    private final FlightDbRepository flightDbRepository;
+    private final AirportService airportService;
+
+    public FlightDbService(FlightDbRepository flightDbRepository, AirportService airportService) {
+        this.flightDbRepository = flightDbRepository;
+        this.airportService = airportService;
+    }
+
+    @Override
+    @ResponseStatus(HttpStatus.OK)
+    public PageResult<Flight> searchFlights(SearchFlightRequest searchFlightRequest) {
+        List<Flight> searchedFlights = new ArrayList<>();
+        if (checkIfSearchValid(searchFlightRequest)) {
+            Airport foundAirportFrom = Arrays.stream(airportService.getAirports(searchFlightRequest.getFrom())).findFirst().orElse(null);
```

Comment:
Varbūt šī metode airportService.getAirports ir jāpārsauc par getAirport (vai findAirport) un viņai jau ir jāatgriež Optional<Airport> ? In memory variantā tas ir galīgi vienkārši, kā arī vispār DB variantā šī vietā:
`List<Airport> findAllByCountryContainingIgnoreCaseOrCityContainingIgnoreCaseOrAirportContainingIgnoreCase
`var uzrakstīt, manuprāt, šādi (nevis findAllBy, bet findBy):
`Optional<Airport> findByCountryContainingIgnoreCaseOrCityContainingIgnoreCaseOrAirportContainingIgnoreCase`